05week/account.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+class Account {
+  constructor(accountNumber, owner, transactions) {
+    this.accountNumber = accountNumber;
+    this.owner = owner;
+    this.transactions = [];
+  }
+  
+  // balance(): loop through all the transactions, and return the current balance
+  balance() {
+    let balance = this.transactions
+            .map(function(t) { return t.amount; })
+            .reduce(function(p, c) { return p + c; },0);
+    return balance;
+  }
+
+  // addTransaction(payee, amount); // checks to see if the amount is valid (ie cannot debit more than the balance)
+  addTransaction(payee, amount) {
```

Comment:
I was hoping for the amount on charges to be always communicated as a positive amount for a charge, and negative amount for a chargeback/refund. and the method internally would record that as a negative amount where appropriate.